feature-toggles/feature-toggles-impl/src/test/java/com/duckduckgo/feature/toggles/codegen/ContributesRemoteFeatureCodeGeneratorBucketAssignmentTest.kt

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.feature.toggles.codegen
+
+import android.content.Context
+import androidx.test.platform.app.InstrumentationRegistry
+import com.duckduckgo.appbuildconfig.api.AppBuildConfig
+import com.duckduckgo.appbuildconfig.api.BuildFlavor.PLAY
+import com.duckduckgo.experiments.api.VariantManager
+import com.duckduckgo.feature.toggles.api.FakeToggleStore
+import com.duckduckgo.feature.toggles.api.FeatureExceptions
+import com.duckduckgo.feature.toggles.api.FeatureSettings
+import com.duckduckgo.feature.toggles.api.FeatureToggles
+import com.duckduckgo.privacy.config.api.PrivacyFeaturePlugin
+import dagger.Lazy
+import kotlin.math.abs
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+import org.robolectric.ParameterizedRobolectricTestRunner
+
+private var enables = 1
+private var disables = 1
+
+@RunWith(ParameterizedRobolectricTestRunner::class)
+class ContributesRemoteFeatureCodeGeneratorBucketAssignmentTest(private val testCase: TestCase) {
+
+    private val context: Context = InstrumentationRegistry.getInstrumentation().targetContext.applicationContext
+    private lateinit var testFeature: TestTriggerFeature
+    private val appBuildConfig: AppBuildConfig = mock()
+    private lateinit var variantManager: FakeVariantManager
+
+    @Before
+    fun setup() {
+        variantManager = FakeVariantManager()
+        whenever(appBuildConfig.flavor).thenReturn(PLAY)
+        testFeature = FeatureToggles.Builder(
+            FakeToggleStore(),
+            featureName = "testFeature",
+            appVersionProvider = { appBuildConfig.versionCode },
+            flavorNameProvider = { appBuildConfig.flavor.name },
+            appVariantProvider = { variantManager.getVariantKey() },
+            forceDefaultVariant = { variantManager.saveVariants(emptyList()) },
+        ).build().create(TestTriggerFeature::class.java)
+    }
+
+    @Test
+    fun `test probability`() {
+        fun isWithinRange(): Boolean {
+            if (testCase.run < 9000) return true
```

Comment:
Why do we need this?

Reply:
I'll add a comment in the test, the reason is that when the sample size is small, we can't ensure the bucket distribution is within the right error bands. So I am basically ignoring sample sizes below 9000